presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestEvaluateProjectOnValues.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.spi.plan.ValuesNode;
+import com.facebook.presto.sql.planner.iterative.rule.test.BaseRuleTest;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.SystemSessionProperties.EVALUATE_PROJECT_ON_VALUES;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.node;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.assignment;
+
+public class TestEvaluateProjectOnValues
+        extends BaseRuleTest
+{
+    @Test
+    public void testDoesNotFireOn()
+    {
+        tester().assertThat(new EvaluateProjectOnValues(tester.getMetadata().getFunctionAndTypeManager()))
+                .setSystemProperty(EVALUATE_PROJECT_ON_VALUES, "true")
+                .on(p -> p.project(p.project(p.values(p.getIdAllocator().getNextId(), 2, p.variable("a")),
```

Comment:
it's difficult to tell what this test case is trying to accomplish. Can you write a comment?

Reply:
Sure